src/batch.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react'
+import classNames from 'classnames'
+
+class Batch extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      searching: false,
+      error: null,
+      results: null,
+      query: null
+    }
+  }
+
+  performSearch(query) {
+    if(!query) {
+      return
+    }
+    if (this.state.query === query) {
+      return
+    }
+
+    this.setState({
+      searching: true,
+      query: query
+    })
+
+    this.search(
+      query,
+      this.onResults.bind(this),
+      this.onError.bind(this))
+  }
+
+  search(query, success, error) {
```

Comment:
I think this would better be named `batchSearch`, because it looks like a copy&paste from `search.js`, but isn't.

Reply:
Yep, will fix.